wscript_build.py

```diff
@@ -580,12 +580,16 @@ def swift(task):
         syms = True
         ctx.load("syms")
 
+    additonal_objects = ""
+    if ctx.dependency_satisfied('swift'):
+        additonal_objects = "osdep/macOS_swift.o"
+
     if ctx.dependency_satisfied('cplayer'):
         ctx(
             target       = "mpv",
             source       = main_fn_c,
             use          = ctx.dependencies_use() + ['objects'],
-            add_object   = "osdep/macOS_swift.o",
+            add_object   = additonal_objects,
```

Comment:
Nit: using plural (object**s**), but content is a single string. See if this works with a list and then just append() to it in the if check in case it works.